### ISSUE TITLE
(F) Бесполезный класс

### ISSUE THREAD

**Issue by G0ld3nDr4g0n3301** [kosnvny/absolutely_new_lab3#14]
Я понимаю,что было тяжело сделать иначе. Но у тебя класс Шурупчик ВООБЩЕ ничем не отличается от Person,кроме метода в пять строк(который нажатие кнопок). Опять же - технически это не ошибка, но я бы добавил больше логики и уникальности в Шурупа. Например добавил бы ему типов передвижения,сделал бы,чтобы он отшатывался в сторону,когда внезапно выпрыгивает стул. Сделал бы ему рандомную механику пульса при испуге, и добавил бы шанс смерти от инфаркта(кстати, это тебе было бы полезно для эксепшонов в следующей лабе), что угодно.В целом - не ошибка,но Максим не любит бесполезных классов.

**Comment by Vediusse** [comment]
он не бесполезный - он предопределяет один контракт + добавляет метод push 


а вот классы basement и scientistroom лишний

**Comment by G0ld3nDr4g0n3301** [comment]
> он не бесполезный - он предопределяет один контракт + добавляет метод push
> 
> а вот классы basement и scientistroom лишний

Согласен,просто для этих целей подошёл бы анонимный класс,так как мы шурупа лишь единожды юзаем,а логика в нём настолько проста,что уместится в анонимный класс. Цельный нам создавать незачем. Но это скорее философский вопрос. Рекомендацию по инфаркту всё же считаю неплохой,так что закрывать ишью не буду.

**Comment by Vediusse** [comment]
<img width="613" alt="Снимок экрана 2024-01-07 в 21 36 04" src="https://github.com/kosnvny/absolutely_new_lab3/assets/109961426/c7da628b-8099-4983-bafb-5123cfe5d3bc">

Анонимный класс будет в 4 классе 

Максим меня заставил сделать такой класс - поэтому в рамках работы всё гут

**Comment by G0ld3nDr4g0n3301** [comment]
Не совсем понял,зачем ему трижды принимать имя? Что там такое в конструкторе хумана?

**Comment by Vediusse** [comment]
https://github.com/Vediusse/lab3_solid/blob/lab4/Lab3/Entity/Human.java

Можешь чекнуть - не для слабонервных

**Comment by G0ld3nDr4g0n3301** [comment]
> https://github.com/Vediusse/lab3_solid/blob/lab4/Lab3/Entity/Human.java
> 
> Можешь чекнуть - не для слабонервных

Ахахах, я вижу ты следуешь трендам? 72 гендера, самоидентификация единорогом-лягушкой-боевым хеликоптером и всё такое))) на самом деле идея динамического изменения текста под pronoun интересная, но её тяжеловато реализовать,так как надо менять всякие "сделал\сделало" и т.д., сложно это.

**Comment by Vediusse** [comment]
> > https://github.com/Vediusse/lab3_solid/blob/lab4/Lab3/Entity/Human.java
> > Можешь чекнуть - не для слабонервных
> 
> Ахахах, я вижу ты следуешь трендам? 72 гендера, самоидентификация единорогом-лягушкой-боевым хеликоптером и всё такое))) на самом деле идея динамического изменения текста под pronoun интересная, но её тяжеловато реализовать,так как надо менять всякие "сделал\сделало" и т.д., сложно это.

На самом деле не так сложно просто сделать subjectBilder а там работа с массивами и с свойствами объектов-элемментов массива )
